quickwit-metastore/src/lib.rs

```diff
@@ -59,3 +63,86 @@ pub use metastore::{
     IndexMetadata, MetadataSet, Metastore, SplitMetadata, SplitMetadataAndFooterOffsets, SplitState,
 };
 pub use metastore_resolver::{MetastoreFactory, MetastoreUriResolver};
+
+/// A trait to allow for our data model to check itself
+#[async_trait]
+pub trait Check {
+    /// Something that can be implemented
+    fn description(&self) -> String;
+
+    /// An async function that implements the check on this type
+    async fn check(&self) -> Result<(), anyhow::Error>;
```

Comment:
I wonder if anyhow::Error is the best type for reporting.

Reply:
The idea here was a minimal implementation, and since the checks touch various parts of the system with different error types, `anyhow::Error` looked like the least common denominator.

Do you have a suggestion what to use instead?

Reply:
I checked the doc of anyhow::Error. It might be nice actually. It is really meant to be able to display different amount of context. Display is probably what we want.

Reply:
I see you that for storage, you introduced a `check` method in the Storage trait and for metastore you introduce an independent Check trait. I am missing the reason for the different approach?

If both works, I like the 

check method in the Storage trait.
check method in the Metastore trait.
and possibly a check_index(&self) method in the Metastore trait.
etc.